src/tools/compiletest/src/main.rs

```diff
@@ -1,4 +1,5 @@
 #![crate_name = "compiletest"]
+#![feature(stmt_expr_attributes)]
```

Comment:
Per the comment just below, you don't want to add new feature gates to compiletest.

Reply:
@Mark-Simulacrum  
The feature is removed, and the rustc complain about this and suggest I put this feature back.
https://github.com/rust-lang/rust/runs/4787922801?check_suite_focus=true#step:25:2113

By the way, may you help me to correct written this?
https://github.com/rust-lang/rust/pull/92714/files#diff-bf53d26b9ca67283fc825e902a64780b463c153d1b1df0c8e782da0300f08956R889

Reply:
So, we do need to avoid the use of stmt_expr_attributes -- that likely needs to happen by removing this feature gate, and then adjusting the code to avoid its use. That may be relatively annoying, but we don't want compiletest using unstable features outside libtest (and ideally not even libtest, but that's not currently really avoidable).

As I noted previously that's likely a matter of putting `{ ... }` blocks around most expressions and cfg'ing those.